src/_pytest/fixtures.py

```diff
@@ -986,7 +1008,40 @@ def __call__(self, function):
         return function
 
 
-def fixture(scope="function", params=None, autouse=False, ids=None, name=None):
+FIXTURE_ARGS_ORDER = ("scope", "params", "autouse", "ids", "name")
+
+
+def _parse_fixture_args(callable_or_scope, *args, **kwargs):
+    arguments = dict(scope="function", params=None, autouse=False, ids=None, name=None)
+
+    fixture_function = None
+    if isinstance(callable_or_scope, str):
+        args = list(args)
+        args.insert(0, callable_or_scope)
+    else:
+        fixture_function = callable_or_scope
+
+    positionals = set()
+    for positional, argument_name in zip(args, FIXTURE_ARGS_ORDER):
+        arguments[argument_name] = positional
+        positionals.add(argument_name)
+
+    duplicated_kwargs = {kwarg for kwarg in kwargs.keys() if kwarg in positionals}
+    if duplicated_kwargs:
+        raise TypeError(
+            "The fixture arguments are defined as positional and keyword: {}. "
+            "Use only keyword arguments.".format(", ".join(duplicated_kwargs))
+        )
+
+    if positionals:
+        warnings.warn(FIXTURE_POSITIONAL_ARGUMENTS, stacklevel=2)
+
+    arguments.update(kwargs)
+
+    return fixture_function, arguments
+
+
+def fixture(callable_or_scope=None, *args, **kwargs):
```

Comment:
@RonnyPfannschmidt, @nicoddemus - I have second thoughts about this part. I think it is not good that the argument names are lost from the function signature. You think that changing `**kwargs` to `scope="function", params=None, autouse=False, ids=None, name=None` will make sense here? It would preserve the IDE's ability to tell which arguments are accepted.

Reply:
we should set a signature, we cant implement the proper warning behaviour otherwise

Reply:
OK, I'll work on it today then.

Reply:
Done.